docker/docker_image_dest.go

```diff
@@ -394,11 +394,13 @@ func (d *dockerImageDestination) TryReusingBlobWithOptions(ctx context.Context,
 			continue
 		}
 
-		return true, private.ReusedBlob{
-			Digest:               candidate.Digest,
-			Size:                 size,
-			CompressionOperation: compressionOperation,
-			CompressionAlgorithm: compressionAlgorithm}, nil
+		if options.RequiredCompression == nil || (options.RequiredCompression == compressionAlgorithm) {
```

Comment:
This check should happen before making any on-registry operations (`blobExists`/`mountBlob`)

Reply:
Done.